src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -61,14 +61,18 @@ function renderIntoDom(reactElement, domElement, errorCount = 0) {
   );
 }
 
+async function renderIntoString(reactElement, errorCount = 0) {
+  return await expectErrors(
+    () => new Promise(resolve => resolve(ReactDOMServer.renderToString(reactElement))),
```

Comment:
Yeah, it behaves differently if `renderToString` throws an error. With `Promise.resolve()`, the arrow function throws, whereas with `new Promise`, it just rejects. There are some tests that I haven't yet submitted that depend on the promises rejecting on error rather than throwing up the call stack.